tests/integration_tests/security/test_vulnerabilities.py

```diff
@@ -400,88 +181,71 @@ def check_vulnerabilities_files_on_guest(microvm):
+            results.append({"file": vuln_file, "stdout": stdout})
+    return results
 
-def check_vulnerabilities_files_ab(builder):
-    """Does an A/B test on the contents of the /sys/devices/system/cpu/vulnerabilities files in the guest if
-    running in a PR pipeline, and otherwise calls `check_vulnerabilities_files_on_guest`
-    """
-    if is_pr():
-        precompiled_ab_test_guest_command(
-            builder,
-            f"! grep -r Vulnerable {VULN_DIR}",
-            comparator=set_did_not_grow_comparator(
-                lambda output: set(output.stdout.splitlines())
-            ),
-        )
-    else:
-        check_vulnerabilities_files_on_guest(builder())
-
-
-def test_vulnerabilities_files_on_guest(build_microvm):
-    """
-    Test vulnerabilities files on guest.
-    """
-    check_vulnerabilities_files_ab(build_microvm)
-
-
-def test_vulnerabilities_files_on_restored_guest(build_microvm, microvm_factory):
-    """
-    Test vulnerabilities files on a restored guest.
-    """
-    check_vulnerabilities_files_ab(with_restore(build_microvm, microvm_factory))
-
-
-def test_vulnerabilities_files_on_guest_with_template(build_microvm_with_template):
-    """
-    Test vulnerabilities files on guest with CPU template.
-    """
-    check_vulnerabilities_files_ab(build_microvm_with_template)
 
-
-def test_vulnerabilities_files_on_guest_with_custom_template(
-    build_microvm_with_custom_template,
-):
-    """
-    Test vulnerabilities files on guest with a custom CPU template.
-    """
-    check_vulnerabilities_files_ab(build_microvm_with_custom_template)
+@pytest.fixture
+def microvm_factory_a(record_property):
+    """MicroVMFactory using revision A binaries"""
+    revision_a = global_props.buildkite_revision_a
+    bin_dir = git_clone(Path("../build") / revision_a, revision_a).resolve()
+    fc_bin = bin_dir / "firecracker"
+    jailer_bin = bin_dir / "jailer"
+    record_property("firecracker_bin", str(fc_bin))
+    uvm_factory = MicroVMFactory(fc_bin, jailer_bin)
+    yield uvm_factory
+    uvm_factory.kill()
 
 
-def test_vulnerabilities_files_on_restored_guest_with_template(
-    build_microvm_with_template, microvm_factory
-):
-    """
-    Test vulnerabilities files on a restored guest with a CPU template.
-    """
-    check_vulnerabilities_files_ab(
-        with_restore(build_microvm_with_template, microvm_factory)
-    )
+@pytest.fixture
+def uvm_any_a(microvm_factory_a, uvm_fun, guest_kernel, rootfs, cpu_template_any):
+    """Return uvm with revision A firecracker"""
+    return uvm_fun(microvm_factory_a, guest_kernel, rootfs, cpu_template_any)
+
+
+def test_check_vulnerability_files_ab(request, uvm_any):
+    """Test vulnerability files on guests"""
+    res_b = check_vulnerabilities_files_on_guest(uvm_any)
+    if global_props.buildkite_pr:
+        # we only get the uvm_any_a fixtures if we need it
+        uvm_a = request.getfixturevalue("uvm_any_a")
```

Comment:
Yes, `guest_kernel` for example will be the same for `uvm_any` and `uvm_any_a`, because pytest caches the fixtures for a test. That's an important property. I will add it to the docstring.

https://docs.pytest.org/en/stable/how-to/fixtures.html#fixtures-can-be-requested-more-than-once-per-test-return-values-are-cached